### DISCUSSION TITLE
MetadataError: Metadata for entity not found

### DISCUSSION THREAD

**Discussion by araines** [mikro-orm/mikro-orm#3387] [Q&A] [answered]
I've been trying to get this working for a few hours now, but I'm no closer to understanding what I'm doing wrong.  Hoping someone can help point me in the right direction!

I have two entities, with a OneToMany relationship.  I'd like to be able to add all the items in to the collection on construction, but when I try I get a `MetadataError` issue.

Simplified version of what I'm trying to do:
```
@Entity()
export default class PullRequestCommit {
  @PrimaryKey()
  id: number

  @ManyToOne(() => PullRequest)
  pullRequest: PullRequest

  @Property()
  sha: string
}

@Entity()
export default class PullRequest {
  @PrimaryKey()
  id: number

  @OneToMany(() => PullRequestCommit, (pullRequestCommit) => pullRequestCommit.pullRequest)
  commits: Collection<PullRequestCommit> = new Collection<PullRequestCommit>(this)

  constructor(commits: PullRequestCommit[]) {
    this.commits.add(...commits)
  }
}

const commit = new PullRequestCommit()
const pr = new PullRequest([commit]) // throws MetadataError: Metadata for entity PullRequest not found
```

**Comment by B4nan** [comment, marked as answer]
You need to initialize the orm first to be able to work with collections.

**Reply by araines** [reply]
Aw man, I knew it would be something criminally simple.  My test suite wasn't initialising things.  Life saver as always, thank you!

**Reply by B4nan** [reply]
The error message could be indeed better for this case. I would rather validate this instead of a fallback (which would be not [propagating changes](https://mikro-orm.io/docs/propagation) in case metadata is missing).
